crates/cext/src/transpiler/transpile_function.rs

```diff
@@ -0,0 +1,181 @@
+    /// The optimization level to run the transpiler with. Valid values are 0, 1, 2, or 3.
+    optimization_level: u8,
+    /// The seed for the transpiler. If set to a negative number this means no seed will be
+    /// set and the RNGs used in the transpiler will be seeded froms system entropy.
+    seed: i64,
+    /// The approximation degree a heurstic dial where 1.0 means no approximation (up to numerical
+    /// tolerance) and 0.0 means the maximum approximation. A `NAN` value indicates that
+    /// approximation is allowed up to the reported error rate for an operation in the target.
+    approximation_degree: f64,
+}
+
+impl Default for TranspileOptions {
+    fn default() -> Self {
+        TranspileOptions {
+            optimization_level: 2,
+            seed: -1,
+            approximation_degree: 1.0,
+        }
+    }
+}
+
+/// @ingroup QkTranspiler
+///
+/// Generate transpiler options defaults
+///
+/// This function generates a QkTranspileOptions with the default settings
+/// This currently is ``optimization_level`` 2, no seed, and no approximation.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub extern "C" fn qk_transpiler_default_options() -> TranspileOptions {
+    TranspileOptions::default()
+}
+
+/// @ingroup QkTranspiler
+/// Transpile a single circuit that was constructed using the C API
+///
+/// The Qiskit transpiler is a quantum circuit compiler that rewrites a given
+/// input circuit to match the constraints of a QPU and/or optimize the circuit
+/// for execution.
+///
+/// This function is multithreaded internally and will launch a thread pool
+/// with threads equal to the number of CPUs by default. You can tune the
+/// number of threads with the ``RAYON_NUM_THREADS`` environment variable.
+/// For example, setting ``RAYON_NUM_THREADS=4`` would limit the thread pool
+/// to 4 threads.
+///
+/// @param circuit A pointer to the circuit to run the transpiler on
+/// @param target A pointer to the target to compile the circuit for
+/// @params options A pointer to an options object that define user options if this is a null
+///   pointer the default values will be used
+/// @param result A pointer to the memory location of the transpiler result. On a successful
+///   execution (return code 0) the output of the transpiler will be written to the pointer
+/// @param error A pointer to a pointer with an nul terminated string with an error description.
+///   If the transpiler fails a pointer to the string with the error description will be written
+///   to this pointer. That pointer needs to be freed with `qk_str_free`.
+///
+/// @returns the return code for the transpiler, 0 means success and all other values indicate an
+///   error
+///
+/// # Safety
+///
+/// Behavior is undefined if ``circuit``, ``target``, ``options``, ``result``, or ``error`` are
+/// not valid, non-null pointers to a ``QkCircuit``, ``QkTarget``, ``QkTranspileOptions``,
+/// ``QkTranspileResult``, or a ``char`` pointer respectively.
```

Comment:
It should be a character pointer pointer, right?
```suggestion
/// ``QkTranspileResult``, or a ``char**`` respectively.
```

Reply:
The PR text is correct - `error` must be a (valid, non-null pointer) to a (`char` pointer).

(Though arguably, we could easily handle the case that it _is_ a null pointer and just not write out the error message.)